diff --git a/fboauth.install b/fboauth.install
index 3043847..fb04a4a 100644
--- a/fboauth.install
+++ b/fboauth.install
@@ -62,6 +62,8 @@ function fboauth_schema() {
  */
 function fboauth_schema_alter(&$schema) {
   $schema['users']['fields']['mail']['length'] = 320;
+  $schema['users']['fields']['mail']['description'] = "User's e-mail address.";
+  $schema['users']['indexes']['mail'] = array(array('mail', 255));
 }
 
 /**
@@ -77,6 +79,8 @@ function _fboauth_change_user_mail_field($action = 'install') {
   if ($action == 'install') {
     fboauth_schema_alter($schema);
   }
-  db_change_field('users', 'mail', 'mail', $schema['users']['fields']['mail']);
+  db_drop_index('users', 'mail');
+  db_change_field('users', 'mail', 'mail', $schema['users']['fields']['mail'],
+    array('indexes' => array('mail' => array(array('mail', 255)))));
   return $res;
 }
